core/model/modx/processors/element/propertyset/updatefromelement.class.php

```diff
@@ -9,44 +10,39 @@
 
 class modPropertySetUpdateFromElementProcessor extends modPropertySetUpdateProcessor {
     public $languageTopics = array('propertyset', 'category', 'element');
-
-    /** @var modPropertySet|null */
-    public $object;
-    /** @var modElement|null  */
-    public $element = null;
-
-    /**
-     * Get element, if necessary
-     * @return null|modElement
-     */
-    public function getElement() {
-        $elementId = (int)$this->getProperty('elementId', 0);
-        $elementClass = $this->getProperty('elementType', '');
-        if ($elementId && !empty($elementClass)) {
-            return $this->modx->getObject($elementClass, $elementId);
-        }
-
-        return null;
-    }
+ 
 
     /**
      * {@inheritdoc}
      * @return bool|null|string
      */
     public function initialize() {
-        $this->element = $this->getElement();
-
-        $primaryKey = $this->getProperty($this->primaryKeyField, false);
-        if ($primaryKey == 'Default') {
-            $primaryKey = 0;
-            $this->setProperty($this->primaryKeyField, 0);
-        }
-
-        if (!$primaryKey) {
-            if (!$this->element) return $this->modx->lexicon('element_err_ns');
-            return true;
+        
+        /*
+             Determine whether the transferred parameter set ID.
+             If so, then the class of the current object left without changes modPropertySet
+             If not, change the class of the class of the updated element.
+             That is the current element $this->object as the case may be, or modPropertySet or modTemplate | modSnippet etc.
+        */
+        
+        if(!$id = (int)$this->getProperty($this->primaryKeyField)){
+            if(!$elementType = $this->getProperty('elementType')){
+                return "Item class has not been specified";
+            }
+            
+            // else
+            $this->classKey = $elementType;
+            $id = (int)$this->getProperty('elementId');
+            
+            // switch($this->classKey){
```

Comment:
Remove unnecessary comments like
`// else`
`// switch($this->classKey){`

etc.